src/Utils/samples.jl

```diff
@@ -344,6 +344,85 @@ function sample!(D::Vector{VN}, X::LazySet, sampler::CombinedSampler;
     throw(ErrorException("sampling failed"))
 end
 
+"""
+    FaceSampler <: AbstractSampler
+
+Type used for sampling from the `k`-faces of a set.
+
+### Fields
+
+- `dim` -- dimension of the faces to be sampled; a negative number is
+           interpreted as `n - dim` where `n` is the dimension of the set
+
+### Notes
+
+For a three-dimensional polytope, the following face dimensions exist:
+- 3-face – the polytope itself
+- 2-faces – 2-dimensional polygonal faces
+- 1-faces – 1-dimensional edges
+- 0-faces – 0-dimensional vertices
+
+For more information see
+[Wikipedia](https://en.wikipedia.org/wiki/Face_(geometry)#k-face).
+"""
+struct FaceSampler <: AbstractSampler
+    dim::Int
+end
+
+function sample!(D::Vector{VN}, X::LazySet, sampler::FaceSampler;
+                 rng::AbstractRNG=GLOBAL_RNG,
+                 seed::Union{Int, Nothing}=nothing) where {N, VN<:AbstractVector{N}}
+    n = dim(X)
+    k = sampler.dim
+
+    # translate negative dimension
+    if k < 0
+        k += n
+    end
+
+    # full-dimensional sampling
+    if k == n
+        return sample!(D, X, _default_sampler(X); rng=rng, seed=seed)
+    end
+
+    0 <= k < n || throw(ArgumentError("cannot sample from " *
+        "$(sampler.dim)-dimensional faces for a set of dimension $n"))
+
+    rng = reseed(rng, seed)
+    return _sample_faces!(D, X, rng, k)
+end
+
+_choose_sorted(k, n; rng=GLOBAL_RNG) = sort!((1:n)[randperm(rng, end)][1:k])
+
+function _sample_faces!(D::Vector{VN}, H::AbstractHyperrectangle, rng, k) where {N, VN<:AbstractVector{N}}
+    n = dim(H)
+    b = 1:2
+
+    @inbounds for j in 1:length(D)
+        indices = _choose_sorted(k, n; rng=rng)
```

Comment:
i think that the algorithm can be optimized by re-using the `indices` vector. that should be useful in the case that D is "very" large..